src/components/Filters/DateRangeFilter.tsx

```diff
@@ -0,0 +1,59 @@
+import { Key } from "react";
+import { Matcher } from "react-day-picker";
+import { BaseFilter } from "src/components/Filters/BaseFilter";
+import { Filter } from "src/components/Filters/types";
+import { Label } from "src/components/Label";
+import { DateRangeField, Value } from "src/inputs";
+import { DateRange } from "src/types";
+import { TestIds } from "src/utils";
+import { defaultTestId } from "src/utils/defaultTestId";
+
+export type DateRangeFilterProps<V extends Value, DV extends DateRangeFilterValue<V>> = {
+  label: string;
+  defaultValue?: DV;
+  placeholderText?: string;
+  disabledDays?: Matcher | Matcher[];
+  // For storybook to support showing dateRange and date filters in same Filter component
+  testFieldLabel?: string;
+};
+
+// Using op to align with what is expected from DateFilter at this time.
+// In the future we should remove that coupling and rely on user to map returned value
+export type DateRangeFilterValue<V extends Value> = { op: V; value: DateRange | undefined };
```

Comment:
Thought: Hm, kind of wondering...you mention we have `op` b/c the backend `DateFilter2` has it, which is true, but the backend type looks like:

```
export interface DateFilter2 {
  op: DateOperation;
  value: Date;
  value2?: Date | null | undefined;
}
```

So it also doesn't use `value: DateRange`, but instead uses two `value` + `value2` `Date`s...

This makes me think the page using this filter is probably going to have to manually map this value anyway, and so maybe we should just skip taking `op` all together (since the rationale of 'matching the backend' isn't 100% right anyway? Afaiu?)

Reply:
I had gone back forth on that as well. When i ran this branch in blueprint, our GQL types in `JobLogsPage` really hated leaving out `op` in the filterDef and we were already mapping our usages of `dateFilter` in the same way with `value1` so i figured no harm in keeping it around _for now._ Removing it will just require adding a little more logic to make TS happy when the filter is used. 

If we remove `op` i'd like to make sure we track this tech debt somewhere to see about adding a new GQL resolver or util for all of our dates and cut out these `} as DateFilter2` for a hack day or something if you agree....that'd just go into the confluence page correct?